src/client/handlers/VaultsSecretsMkdir.ts

```diff
@@ -2,50 +2,60 @@ import type { DB } from '@matrixai/db';
 import type {
   ClientRPCRequestParams,
   ClientRPCResponseResult,
-  SecretMkdirMessage,
-  SuccessMessage,
+  SecretDirMessage,
+  SuccessWithErrorMessage,
 } from '../types';
 import type VaultManager from '../../vaults/VaultManager';
-import { UnaryHandler } from '@matrixai/rpc';
+import { DuplexHandler } from '@matrixai/rpc';
 import * as vaultsUtils from '../../vaults/utils';
 import * as vaultsErrors from '../../vaults/errors';
 import * as vaultOps from '../../vaults/VaultOps';
 
-class VaultsSecretsMkdir extends UnaryHandler<
+class VaultsSecretsMkdir extends DuplexHandler<
   {
     vaultManager: VaultManager;
     db: DB;
   },
-  ClientRPCRequestParams<SecretMkdirMessage>,
-  ClientRPCResponseResult<SuccessMessage>
+  ClientRPCRequestParams<SecretDirMessage>,
+  ClientRPCResponseResult<SuccessWithErrorMessage>
 > {
-  public handle = async (
-    input: ClientRPCRequestParams<SecretMkdirMessage>,
-  ): Promise<ClientRPCResponseResult<SuccessMessage>> => {
-    const { vaultManager, db } = this.container;
-    await db.withTransactionF(async (tran) => {
-      const vaultIdFromName = await vaultManager.getVaultId(
-        input.nameOrId,
-        tran,
-      );
-      const vaultId =
-        vaultIdFromName ?? vaultsUtils.decodeVaultId(input.nameOrId);
-      if (vaultId == null) {
-        throw new vaultsErrors.ErrorVaultsVaultUndefined();
-      }
-      await vaultManager.withVaults(
-        [vaultId],
-        async (vault) => {
-          await vaultOps.mkdir(vault, input.dirName, {
-            recursive: input.recursive,
-          });
-        },
-        tran,
-      );
-    });
-    return {
-      success: true,
-    };
+  public handle = async function* (
+    input: AsyncIterable<ClientRPCRequestParams<SecretDirMessage>>,
+  ): AsyncGenerator<ClientRPCResponseResult<SuccessWithErrorMessage>> {
+    const { vaultManager, db }: { vaultManager: VaultManager; db: DB } =
+      this.container;
+    let metadata: any = undefined;
```

Comment:
Usually, in TypeScript, the `any` type is avoided, as it bypasses type checking, defeating the purpose of types in the first place. Here, however, we need to use the any type as we are parsing the metadata directly.

Should we somehow simplify this to not include `any` type, or is this usage acceptable?

Reply:
Metadata has a type already that isn't `any`, just reuse that. I think it's `Record<string, JSONValue>` but you'll have to check.

Reply:
I tried this and it didn't work. I checked the type and it is more complex than that, containing multiple unions between types.

Instead, a `JSONObject` type worked. The types were still kinda borked, as my LSP complained that not all optional types have the `options` and `recursive` fields. To fix this, I had to do the following:
```ts
const options = metadata?.options as { recursive?: boolean };
response = await vaultOps.mkdir(vault, dirName, {
  recursive: options.recursive,
});
```
This is kind of a janky solution, but if I don't have the `as type`, then TS will check if the attribute `recursive` exists on all the types, which of course it doesn't, and it will raise an error in the linter.

Reply:
This is not the correct solution, as I am using the `as` keyword here. After having a talk with Brian regarding this, using the `as` keyword in all but the most critical cases is discouraged, so I will find an alternative solution for this.

Reply:
My solution was to use the `POJO` type, as it is defined as `[key: string]: any`. This is not strict type checking, but using the exact type as of the metadata field makes it challenging to access potentially deeply nested fields. The `POJO` type retains some type checking while still allowing the freedom of `any`.

This is not a perfect solution, but this is better than using `any`. These changes will need to be made in other handlers too which are expecting values in `metadata`, but that can be handled in future PRs as this one is already getting pretty large.